Updates for Ubuntu 20.04 (#243)
uyuni-project/uyuni-docs: modules/client-configuration/pages/clients-ubuntu.adoc

However, the root user on {ubuntu} is disabled by default, so to use bootstrapping, you will require an existing user with [command]``sudo`` privileges for Python.

== Prepare to Register

Some preparation is required before you can register {ubuntu} clients to the {productname} Server.

ifeval::[{suma-content} == true]

> .Procedure: Adding client tools channels

Reviewer: ```suggestion
.Procedure: Adding Client Tools Channels
```

* From the {webui}, add [systemitem]``Ubuntu 18.04`` and [systemitem]``SUSE Linux Enterprise Client Tools Ubuntu 1804 amd64``.
* From the command prompt, add [systemitem]``ubuntu-18.04-pool-amd64`` and [systemitem]``ubuntu-18.04-suse-manager-tools-amd64``.

For {ubuntu} 20.04:

* From the {webui}, add [systemitem]``Ubuntu 20.04`` and [systemitem]``SUSE Linux Enterprise Client Tools Ubuntu 2004 amd64``.
* From the command prompt, add [systemitem]``ubuntu-20.04-pool-amd64`` and [systemitem]``ubuntu-20.04-suse-manager-tools-amd64``.

[NOTE]
====

> The mandatory channels do not contain {ubuntu} upstream packages.

Reviewer 2: This is incorrect.

> The mandatory channels do not contain {ubuntu} upstream packages.

Reviewer 2: This is not correct anymore.

Author: ![image](https://user-images.githubusercontent.com/3914967/83858447-dece1180-a75f-11ea-892c-987addd80317.png)

Reviewer 2: Click on the "bars" for "Ubuntu 20.04". You will find the channels there.

Author: So it needs to be all three of the Mandatory ones here? 
![image](https://user-images.githubusercontent.com/3914967/83859523-65372300-a761-11ea-93c6-6addc90f185a.png)

Reviewer 2: Yes, but as you can see on the WebUI, you can't pick them one by one (you check the product, and that's it).

In the {productname} {webui}, navigate to menu:Software[Channel List > All].
Verify that you have a base channel and a child channel for your architecture.

> For example, for {ubuntu} 20.04:

Reviewer 2: I don't understand, why do you have this if you already explained how to add the channels above?

Author: This is a verification step.

Reviewer 2: Can you please generate the rendered doc? At least on the adoc this looks wrong.

* Base channel: [systemitem]``ubuntu-2004-amd64-main for amd64``
* Child channel: [systemitem]``Ubuntu-20.04-SUSE-Manager-Tools for amd64``

////
You need to manually create three repositories:

* For main
* For main updates
* For main security

These examples are for {ubuntu} 20.04 (focal).
Make sure you change the values to match the {ubuntu} version you want to use.

In these procedures, we use ``http://archive.ubuntu.com/`` for {ubuntu} repositories.
You can replace this URL with a more appropriate mirror.

> 

Reviewer 2: We don't need custom channels anymore for Ubuntu 20.04.

We need a mumble about this to clarify. I can't review and review the PR 20 times, so for now I will stop here.

Author: No, I started updating them, then realised and commented them out.

Reviewer 2: as discussed, we don't need them for 20.04, but we need them for the other Ubuntu versions.

* Child custom channel: [systemitem]``Ubuntu 18.04 LTS AMD64 Main Security``
* Child vendor channel: [systemitem]``Ubuntu-18.04-SUSE-Manager-Tools for amd64``

When you have the channels set up, associate each channel with the appropriate repository, and synchronize the channels.
You can also configure a synchronization schedule at this point.

> To synchronize the channels, navigate to menu:Software[Manage > Channels], and for each channel you created, click on it and:

Reviewer: Instead of "click on it" I'd either rather say "click it" or "select it".

Later l.196 we have this variation:

> Click each channel in the list, and:

. Navigate to the [guimenu]``Repositories`` tab.
. Navigate to the [guimenu]``Sync`` tab.
. Click btn:[Sync Now] to begin synchronization immediately.

> . You can also setup a scheduled synchronization from this screen.

Reviewer: ```suggestion
. You can also set up a scheduled synchronization from this screen.
```

====
This procedure grants root access without requiring a password, which is required for registering the client.
When the client is successfully installed it will run with root privileges, so the access is no longer required.
We recommend that you remove the line from the [path]``sudoers`` file after the client has been successfully installed.
====

== Register Clients

To register your {ubuntu} clients, you need a bootstrap repository.
By default, bootstrap repositories are automatically created, and regenerated daily for all synchronized products.
You can manually create the bootstrap repository from the command prompt, using this command:

----

> mgr-create-bootstrap-repo

Reviewer 2: As I said, I am not 100% sure about this. Check with Michael before marging.

Author: @mcalmer can you please advise?

Reviewer 3: for 16.04 and 18.04 you need it. For 20.04 it is not needed

ubuntu-1804-amd64-universe-updates-uyuni \
ubuntu-1804-amd64-uyuni-client
----

For {ubuntu} 20.04:

> +

Reviewer 2: This is list is correct.

Synchronization can sometimes take several hours.
====

When you have the channels set up, associate each channel with the appropriate repository, and synchronize the channels.
You can also configure a synchronization schedule at this point.

> To synchronize the channels, navigate to menu:Software[Manage > Channels], and for each channel you created, click on it and:

Reviewer: "click it"?

You do not need to delete any previously stored keys.

> If you are bootstrapping clients from the {productname} {webui}, you will need to use a salt state to trust the key.

Reviewer: ```suggestion
If you are bootstrapping clients from the {productname} {webui}, you will need to use a Salt state to trust the key.
```

> Create the salt state and assign it to the organization.

Reviewer: ```suggestion
Create the Salt state and assign it to the organization.
```

> We recommend that you remove the line from the `sudoers` file after the client has been successfully installed.

Reviewer: ```suggestion
We recommend that you remove the line from the [path]``sudoers`` file after the client has been successfully installed.
```
